ls1intum/Artemis
# `Programming exercises`: Fix student build plan creation in Jenkins setups

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context


Build plans for students are not working in Jenkins setups. The build plan contains a link to the base exercise template rather than the student repo.

### Description
The URL for the repository was not correctly changed in the Jenkins build plans. The replacement parameters for the old/new-url were passed into the replacement method the wrong way round.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise
- GitLab/Jenkins setup

1. Start a new participation in the exercise as a student.
2. Commit some code.
3. As instructor, go to the scores page for the exercise.
4. In the submission history, the result should be linked to the latest commit rather than the initial commit that was created when setting up the exercise.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage
unchanged
